handlers/secrets_test.go

```diff
@@ -0,0 +1,236 @@
+package handlers
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/swarm"
+	"github.com/openfaas/faas/gateway/requests"
+)
+
+func genFakeSecret(name string, data string, includeOwnerLabel bool) swarm.Secret {
+	secret := swarm.Secret{
+		ID: name,
+		Spec: swarm.SecretSpec{
+			Data: []byte(data),
+			Annotations: swarm.Annotations{
+				Name: name,
+			},
+		},
+	}
+
+	if includeOwnerLabel {
+		secret.Spec.Annotations.Labels = map[string]string{
+			ownerLabel: ownerLabelValue,
+		}
+	}
+
+	return secret
+}
+
+func getInitialSecrets(includeNonOwnedSecrets bool) map[string]swarm.Secret {
+	initialSecrets := map[string]swarm.Secret{
+		"foo":        genFakeSecret("foo", "baz", true),
+		"foobar":     genFakeSecret("foobar", "bar", true),
+		"baz_fuz":    genFakeSecret("baz_fuz", "foo", true),
+		"barfoo_baz": genFakeSecret("barfoo_baz", "foobar", true),
+	}
+
+	if includeNonOwnedSecrets {
+		initialSecrets["without_label"] = genFakeSecret("without_label", "test", false)
+	}
+
+	return initialSecrets
+}
+
+type fakeDockerSecretAPIClient struct {
+	secrets map[string]swarm.Secret
+}
+
+func newFakeDockerSecretAPIClient() fakeDockerSecretAPIClient {
+	return fakeDockerSecretAPIClient{
+		secrets: getInitialSecrets(true),
+	}
+}
+
+func (c *fakeDockerSecretAPIClient) Reset() {
+	c.secrets = getInitialSecrets(true)
+}
+
+func (c *fakeDockerSecretAPIClient) SecretList(
+	_ context.Context,
+	options types.SecretListOptions,
+) ([]swarm.Secret, error) {
+	secrets := []swarm.Secret{}
+
+	for _, value := range c.secrets {
+		secrets = append(secrets, value)
+	}
+
+	return secrets, nil
+}
+
+func (c *fakeDockerSecretAPIClient) SecretCreate(
+	_ context.Context,
+	secretDesc swarm.SecretSpec,
+) (types.SecretCreateResponse, error) {
+	id := secretDesc.Name
+
+	newSecret := swarm.Secret{
+		ID: id,
+		Spec: swarm.SecretSpec{
+			Annotations: swarm.Annotations{
+				Name: secretDesc.Name,
+				Labels: map[string]string{
+					ownerLabel: ownerLabelValue,
+				},
+			},
+			Data: secretDesc.Data,
+		},
+	}
+
+	c.secrets[secretDesc.Name] = newSecret
+
+	return types.SecretCreateResponse{ID: id}, nil
+}
+
+func (c *fakeDockerSecretAPIClient) SecretRemove(_ context.Context, id string) error {
```

Comment:
The indenting is mixed between the various methods, this method is a single line but the rest are multiple line, these should all be the same, ideally.  

Since this is implementing a 3rd party interface that is not likely to change, a single like is probably ok for readability. I am not sure what the accepted standard is, perhaps @openfaas/core  could comment.

Reply:
So about indenting, my rules are, if function declaration takes more than 100 characters, I'm splitting it into multiple lines, but if not, I'm leaving it in one line, but if our style is different I don't have any problems with changing my way of doing that. :-)

Reply:
That is why I asked core to chime in, I have seen different suggestions in the go community and I wasn't sure if we had a preference here

Reply:
@alexellis i don't think we should block the merge on this comment. It otherwise looks good to me

Reply:
Thanks for looking at this @LucasRoesler - I'll merge.